common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java

```diff
@@ -155,22 +157,38 @@ public void saveTokens(
         for (final IShareSingleSignOnState<MicrosoftAccount, MicrosoftRefreshToken> sharedSsoCache : mSharedSSOCaches) {
             sharedSsoCache.setSingleSignOnState(account, refreshToken);
         }
+
+        return null; // Returning null, since the ADAL cache's schema doesn't support this return type.
+    }
+
+    @Override
+    public ISaveTokenResult loadTokens(com.microsoft.identity.common.internal.dto.Account account) {
+        throw new UnsupportedOperationException(
+                ERR_UNSUPPORTED_OPERATION
```

Comment:
I'm not thrilled about to do this a ton, but I don't see a fantastic way to avoid it other that hardwiring the dependency in ADAL to `AdalOAuth2TokenCache` and having it no longer implement this `interface`.

^^ This may not be a half-bad idea, come to think of it. +1 if you agree